scripts/gcloud.sh

```diff
@@ -64,9 +66,23 @@ wait_until_node_is_ready() {
 
 # Get external IP for running node or die
 # $1 - VM name
+# $2 - preemptible
+# NB: If the instance is preemptible, the response will add an extra column which will
+# make awk parse the wrong ip.
+# e.g. w/preemptible
+# NAME                       ZONE            MACHINE_TYPE  PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP  STATUS
+# testing-1635856913-node-1  europe-west6-a  e2-medium     true         10.172.0.23  34.65.88.77  RUNNING
+# e.g. w/o preemptible
+# NAME                       ZONE            MACHINE_TYPE  INTERNAL_IP  EXTERNAL_IP  STATUS
+# testing-1635856913-node-1  europe-west6-a  e2-medium     10.172.0.23  34.65.88.77  RUNNING
 gcloud_get_ip() {
   local vm_name="${1}"
-  gcloud compute instances list | grep "${vm_name}" | awk '{ print $5 }'
+  local preemptible="${2:-}"
```

Comment:
Instead of adding another parameter you could also just grep the output for `PREEMPTIBLE` and act upon that.

Reply:
That's definitely another option. TBH neither are pretty, we should really use something else to get the IP of the VM machines rather than `grep`ing the output of `instances list`, WDYT?